synapse/lib/auth.py

```diff
@@ -375,11 +375,25 @@ async def feedBeholder(self, evnt, info, gateiden=None, logged=True):
 
             await self.fire('cell:beholder', **behold)
 
-    @s_nexus.Pusher.onPushAuto('user:info')
     async def setUserInfo(self, iden, name, valu, gateiden=None, logged=True, mesg=None):
 
         user = await self.reqUser(iden)
 
+        if name == 'locked' and not valu and user.isArchived():
+            raise s_exc.BadArg(mesg='Cannot unlock archived user.', user=iden, username=user.name)
+
+        if name in ('locked', 'archived') and not valu:
+            self.checkUserLimit()
+
+        await self._push('user:info', iden, name, valu, gateiden=gateiden, logged=logged, mesg=mesg)
+
+    @s_nexus.Pusher.onPush('user:info')
+    async def _setUserInfo(self, iden, name, valu, gateiden=None, logged=True, mesg=None):
+        user = await self.reqUser(iden)
+
+        if name == 'locked' and not valu and user.isArchived():
```

Comment:
In an upgraded mirror + old leader configuration, would set of edits with locked=False and then archived=False cause a schism? It would remain locked on the mirror, but unlocked on the leader and I don't think the migration on the leader would catch it.

Reply:
Great catch!

Reply:
I think this is resolved now.